src/systems/follow_actor/FollowActor.cc

```diff
@@ -209,7 +209,9 @@ void FollowActor::PreUpdate(const UpdateInfo &_info,
   this->dataPtr->lastUpdate = _info.simTime;
 
   // Is there a follow target?
-  if (this->dataPtr->targetEntity == kNullEntity)
+  if (this->dataPtr->targetEntity == kNullEntity ||
+    _ecm.HasEntity(this->dataPtr->targetEntity) ||
+    _ecm.HasEntity(this->dataPtr->actorEntity))
```

Comment:
I think this should be:

```suggestion
  if (this->dataPtr->targetEntity == kNullEntity ||
    !_ecm.HasEntity(this->dataPtr->targetEntity) ||
    !_ecm.HasEntity(this->dataPtr->actorEntity))
```